### PR TITLE
hyperv: lookup machine on local filesystem first

### PR DESCRIPTION
**Pull request by baude**
when looking for a machine, look it up locally first to prevent accidental collision with non-podman machine vms.  in the cast of `podman machine ls`, only list podman machines found by json files

#### Does this PR introduce a user-facing change?

```release-note
none

```

### CODE REVIEW

File: pkg/machine/hyperv/machine.go
Comment (vrothberg) [inline pkg/machine/hyperv/machine.go:391]:
```suggestion
			return fmt.Errorf("%q: %w", fqConfigPath, machine.ErrNoSuchVM)
```
Not sure if the path is useful to users but in case it is.

**Comment by vrothberg** [comment]
Needs the magic `[NO NEW TESTS NEEDED]` string in the commit message.

**Review by vrothberg — APPROVED** [review]
LGTM
@containers/podman-maintainers PTAL

**Review by Luap99 — COMMENTED** [review]
one small nit

**Review by Luap99 — COMMENTED** [review]
/lgtm
/hold

**Pull request merged**
